main.py

```diff
@@ -184,16 +162,15 @@ def create_circuit(self):
         # Try to create the circuit object
         data = request.get_json()
 
-        name = data.get('name')
-        uni_a = self._get_uni_from_request(data.get('uni_a'))
-        uni_z = self._get_uni_from_request(data.get('uni_z'))
-        creation_time =  self._get_time_from_data(data.get('creation_time'))
-
+        data = self.deserialize_evc_dict(data)
```

Comment:
we have evc.as_dict, as_json, in another parts of code we also have,  `from_json(data) `. Please rename deserialize_evc_dict, to from_json, or from_dict.